File: LC 828.cpp
class Solution {
public:
    int uniqueLetterString(string s) {
        int n = s.length();
        vector<int> last0(26, -1), last1(26, -1);
        int ans = 0, cnt = 0;
        for (int i = 0; i < n; ++i) {
            int c = s[i] - 'A';
            cnt += i - last0[c] - (last0[c] - last1[c]);
            ans += cnt;
            last1[c] = last0[c];
            last0[c] = i;
        }
        return ans;
    }
};

File: 1000/LC 828.go
func uniqueLetterString(s string) (ans int) {
    var last0, last1 [26]int
    for i := range last0 {
        last0[i], last1[i] = -1, -1
    }
    cnt := 0
    for i, c := range s {
        x := int(c - 'A')
        cnt += i - last1[x] - (last1[x] - last0[x])
        ans += cnt
        last0[x], last1[x] = last1[x], i
    }

    return
}